metrics/config_test.go

```diff
@@ -509,8 +498,8 @@ func TestGetMetricsConfig(t *testing.T) {
 			if err != nil {
 				t.Errorf("Wanted valid config %v, got error %v", test.expectedConfig, err)
 			}
-			if !reflect.DeepEqual(*mc, test.expectedConfig) {
-				t.Errorf("Wanted config %v, got config %v", test.expectedConfig, *mc)
+			if diff := cmp.Diff(test.expectedConfig, *mc, cmp.AllowUnexported(*mc), cmpopts.IgnoreTypes(mc.recorder)); diff != "" {
```

Comment:
Since we're adding an anonymous function here, I think the best we could do would be to validate whether or not `recorder` was set.

I've added a new test, `TestStackdriverRecord`, to the bottom of `config_test.go` for now, but it should be moved to a stackdriver-specific module.